src/viam/sdk/resource/resource_api.hpp

```diff
@@ -44,6 +44,15 @@ class API : public APIType {
     bool is_service_type();
     friend bool operator==(API const& lhs, API const& rhs);
     friend bool operator<(const API& lhs, const API& rhs);
+    friend std::ostream& operator<<(std::ostream& os, const API& v);
+
+    template <typename T>
+    struct api_map;
+
+    template <typename T>
+    static API for_t() {
```

Comment:
I think you could plausibly drop the `_t` here.

Reply:
Possibly just a linter thing, but when I tried to name it `for()` I got an error message (`Expected member name or ';' after declaration specifiers` which I took to mean that `for` is a reserved keyword and so couldn't be used here. Even if it is allowed, the fact that my editor complained about and reformatted it makes me think that having a more qualified name is going to be preferable for avoiding confusion.

Reply:
Oh, of course. It's a keyword then. Sorry. I thought you were trying to echo the `_t` from type trait helper functions (e.g. `std::remove_reference_t`), which I don't think you need, rather than working around it being a keyword. Maybe `API::get<Motor>` or `API::of<Motor>`?

Reply:
Sure, I like `API::get<T>`. Just changed!